ml4h/ml4ht_integration/tensor_map.py

```diff
@@ -0,0 +1,138 @@
+from typing import Callable, List, Tuple
+
+import h5py
+
+from ml4h.TensorMap import TensorMap, Interpretation
+from ml4ht.data.data_description import DataDescription
+from ml4ht.data.defines import SampleID, LoadingOption, Tensor, Batch
+
+
+class TensorMapSampleGetter:
+    def __init__(
+            self,
+            tensor_maps_in: List[TensorMap],
+            tensor_maps_out: List[TensorMap],
+            augment: bool = False,
+            return_path: bool = False,
+    ):
+        """
+        A SampleGetter that is built from TensorMaps.
+        This allows backwards compatibility with old TensorMaps.
+
+        :param tensor_maps_in: tensor maps to use to get input data for a model
+        :param tensor_maps_out: tensor maps to use to get output data for a model
+        :param augment: whether to apply the TensorMaps' data augmentations.
+                        This should be true for training and false for testing.
+
+        Example preparing TensorMaps for use in ml4h model:
+        ```
+        import torch
+
+        from ml4h.tensormap.ecg import ecg_rest_raw
+        from ml4h.tensormap.ecg import ecg_rest_age
+        from ml4h.ml4ht_integration.tensor_map import TensorMapSampleGetter
+        from ml4h.models.model_factory import block_make_multimodal_multitask_model
+
+        from ml4ht.data.data_loader import numpy_collate_fn, SampleGetterDataset
+
+        model = block_make_multimodal_multitask_model(
+            [ecg_rest_raw],
+            [ecg_rest_age],
+            ...
+        )
+
+        sg = TensorMapSampleGetter(
+            [ecg_rest_raw],
+            [ecg_rest_age],
+            True,
+        )
+        dset = SampleGetterDataset(
+            [1_path_to_hd5.hd5, 2_path_to_hd5.hd5, ...],
+            sg,
+        )
+        data_loader = torch.utils.data.DataLoader(
+            dset, batch_size=128, collate_fn=numpy_collate_fn,
+        )
+
+        model.fit(iter(data_loader))  # train for one epoch
```

Comment:
Is this still true, or do you have to make a tensorflow dataset from the data loader?

Reply:
Thats what I did now, but if num_workers is too high I get Queue.empty errors in dataloader